File: FileLogging/LogEntry.vb
Imports Microsoft.Extensions.Logging
Imports ova.Common.Logging.Abstract
Imports ova.Common.UnixTime.Base

Namespace Entry

    Public Class LogEntry ' : Inherits LogEntryAbstract
        'Public Sub New()
        '    MyBase.New
        'End Sub

        Public Sub New()
            TimeStampUnix = Timestamp()
            UserName = Environment.UserName
        End Sub

        Public Shared ReadOnly StaticHostName As String = Net.Dns.GetHostName()

        Public ReadOnly Property HostName As String
            Get
                Return StaticHostName
            End Get
        End Property

        Public Property UserName As String
        Public Property TimeStampUnix As Long
        Public Shadows Property Category As String
        Public Property Level As LogLevel
        Public Property Text As String
        Public Property Exception As Exception
        Public Property EventId As EventId
        Public Property State As Object
        Public Property StateText As String
        Public Property StateProperties As Dictionary(Of String, Object)
        Public Property Scopes As List(Of LogScopeInfo)
    End Class

    Public Class LogScopeInfo ' : Implements ILogScopeInfo
        Public Property Text As String ' Implements ILogScopeInfo.Text
        Public Property Properties As Dictionary(Of String, Object) ' Implements ILogScopeInfo.Properties
    End Class

End Namespace

File: Logging/Logging.Shared/LogEntry.cs
using Microsoft.Extensions.Logging;
using System;
using System.Collections.Generic;

namespace ova.Common.Logging.Shared
{
   public class LogEntry 
    {
        public LogEntry()
        {
            TimeStamp =UnixTime.Base.Timestamp();
            UserName = Environment.UserName;
        }

        static public readonly string StaticHostName = System.Net.Dns.GetHostName();
        public string UserName { get; private set; }
        public string HostName { get { return StaticHostName; } }

        public long TimeStamp { get; private set; }
        public string Category { get; set; }
        public LogLevel Level { get; set; }
        public string Message { get; set; }
        public Exception Exception { get; set; }
        public EventId EventId { get; set; }
        public object State { get; set; }
        public string StateText { get; set; }
        public Dictionary<string, object> StateProperties { get; set; }
        public IList<LogScopeInfo> Scopes { get; set; }
        //IList<ILogScopeInfo> ILogEntry.Scopes { get; set; }
    }
}
